dpks/denoise.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+import numpy.typing as npt
+
+from typing import Optional, Any
+
+from sklearn.linear_model import SGDClassifier
+from sklearn.ensemble import BaggingClassifier
+from sklearn.model_selection import StratifiedKFold
+
+
+class BaggedDenoiser(BaggingClassifier):  # type: ignore
+    def __init__(
+        self,
+        estimator: Optional[Any] = None,
+        n_estimators: int = 250,
+        max_samples: float = 1.0,
+        n_jobs: int = 5,
+        random_state: int = 0,
+    ):
+
+        if not estimator:
+
+            estimator = SGDClassifier(
+                alpha=1e-05,
+                average=True,
+                loss="log_loss",
+                max_iter=500,
+                penalty="l2",
+                shuffle=True,
+                tol=0.0001,
+                learning_rate="adaptive",
+                eta0=0.001,
+                fit_intercept=True,
+                random_state=random_state
+            )
+
+        super().__init__(
+            estimator=estimator,
+            n_estimators=n_estimators,
+            max_samples=max_samples,
+            bootstrap=True,
+            n_jobs=n_jobs,
+            random_state=random_state,
+        )
+
+    def vote(self, X: np.ndarray, y: np.ndarray, threshold: float = 0.5) -> np.ndarray:
```

Comment:
Why do we take a threshold here if 0.5 is always used (theres no way to change the parameters in vote() in predict().